MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -460,16 +460,7 @@ public string GetInputArgument(EncodingJobInfo state, EncodingOptions encodingOp
             if (state.IsVideoRequest
                 && string.Equals(encodingOptions.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase))
             {
-                var hasGraphicalSubs = state.SubtitleStream != null && !state.SubtitleStream.IsTextSubtitleStream && state.SubtitleDeliveryMethod == SubtitleDeliveryMethod.Encode;
-                var hwOutputFormat = "vaapi";
-
-                if (hasGraphicalSubs)
-                {
-                    hwOutputFormat = "yuv420p";
-                }
```

Comment:
To pair with line 1633 ~ 1677, use vaapi instead of yuv420p otherwise the transcoding will fail.

Reply:
But it was added there for a reason... I'm wary of changing pieces I don't completely understand. Do you know why it was added there in the first place?

Reply:
Because whether to burn in graphical subtitles or not, it is necessary to use `vaapi` as the hardware output format when using `h264_vaapi` encoder. `yuv420p` is a format used by `libx264` software encoder, so it will cause unsupport and errors for vaapi. Similarly, when using `libx264` to burn graphical subtitles, `-pix_fmt yuv420p` is required.